alvr/server/src/connection.rs

```diff
@@ -698,53 +697,67 @@ fn connection_pipeline(
         let client_hostname = client_hostname.clone();
         thread::spawn(move || {
             while is_streaming(&client_hostname) {
-                let device = match AudioDevice::new_output(
-                    Some(settings.audio.linux_backend),
-                    config.device.as_ref(),
-                ) {
-                    Ok(data) => data,
-                    Err(e) => {
-                        warn!("New audio device failed: {e:?}");
-                        thread::sleep(RETRY_CONNECT_MIN_INTERVAL);
-                        continue;
-                    }
-                };
-
-                #[cfg(windows)]
-                if let Ok(id) = alvr_audio::get_windows_device_id(&device) {
-                    ctx.events_queue
-                        .lock()
-                        .push_back(ServerCoreEvent::SetOpenvrProperty {
-                            device_id: *alvr_common::HEAD_ID,
-                            prop: alvr_session::OpenvrProperty::AudioDefaultPlaybackDeviceId(id),
-                        })
-                } else {
-                    continue;
-                };
-
-                if let Err(e) = alvr_audio::record_audio_blocking(
+                #[cfg(target_os = "linux")]
+                if let Err(e) = alvr_audio::linux::record_audio_blocking_pipewire(
                     Arc::new({
                         let client_hostname = client_hostname.clone();
                         move || is_streaming(&client_hostname)
                     }),
                     game_audio_sender.clone(),
-                    &device,
                     2,
-                    config.mute_when_streaming,
+                    game_audio_sample_rate,
                 ) {
                     error!("Audio record error: {e:?}");
                 }
 
-                #[cfg(windows)]
-                if let Ok(id) = AudioDevice::new_output(None, None)
-                    .and_then(|d| alvr_audio::get_windows_device_id(&d))
+                #[cfg(not(target_os = "linux"))]
                 {
```

Comment:
I think it's better if you leave some spaces within this block. When you have a if block and then another if block, and before `#[cfg()]`